bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs/diff/Vcs.scala

```diff
@@ -0,0 +1,159 @@
+package edu.gemini.sp.vcs.diff
+
+import edu.gemini.pot.sp.{ISPFactory, ISPProgram}
+import edu.gemini.pot.sp.version._
+import edu.gemini.shared.util.VersionComparison.{Same, Newer}
+import edu.gemini.sp.vcs.diff.ProgramLocation.{LocalOnly, Neither, Remote}
+import edu.gemini.sp.vcs.diff.VcsFailure.{NeedsUpdate, VcsException}
+import edu.gemini.sp.vcs.log.VcsEventSet
+import edu.gemini.spModel.core.{Peer, SPProgramID}
+import edu.gemini.util.security.auth.keychain.KeyChain
+
+import java.security.Principal
+
+import scala.collection.JavaConverters._
+import scalaz._
+import Scalaz._
+
+import Vcs._
+
+/** Vcs provides the public API for vcs commands such as push, pull and sync. */
+class Vcs(kc: KeyChain, server: VcsServer) {
+
+  private def user: Set[Principal] = kc.subject.getPrincipals.asScala.toSet
```

Comment:
I would lift this into `VcsAction` because it's a side-effecting action.

```scala
private val user: VcsAction[Set[Principal]] = Task.delay(...).liftVcs
```